### ISSUE TITLE
[Source Request] Animepahe source

### ISSUE THREAD

**Issue by Laezor** [zyrouge/yukino-app#61]
URL: https://animepahe.com
Type: Anime
Language: English

The site includes fansub groups in the anime title.

**Comment by zyrouge** [comment]
Sadly animepahe requests some headers which cant be set due to technical limitations unless i find a way to play m3u8 streams by manually parsing it.

**Comment by Laezor** [comment]
If I am not wrong vlc player can player m3u8 streams.

**Comment by zyrouge** [comment]
> If I am not wrong vlc player can play m3u8 streams.

yes vlc can play m3u8 streams but cant set http headers in vlc, so no way around :(

**Comment by Laezor** [comment]
what about this player? https://dplayer.js.org/guide.html#hls

**Comment by zyrouge** [comment]
> what about this player? https://dplayer.js.org/guide.html#hls

`Referer` header cannot be set manually as browsers restrict it, so have to find a way to do it from the background process (main process/native http)

**Comment by zyrouge** [comment]
Seems like mpv player supports custom headers but using a third party app on a anime "streaming" app doesnt seem nice.

**Comment by Laezor** [comment]
I see, yeah indeed it doesn't seem nice. Maybe this can help https://github.com/open-dynaMIX/simple-mpv-webui. Otherwise, I can suggest another source. Which provides access to google drive links, that should work, right?

**Comment by zyrouge** [comment]
> I see, yeah indeed it doesn't seem nice. Maybe this can help https://github.com/open-dynaMIX/simple-mpv-webui. Otherwise, I can suggest another source. Which provides access to google drive links, that should work, right?

No scraping google drive links aren't easy.

**Comment by ArjixWasTaken** [comment]
> > what about this player? https://dplayer.js.org/guide.html#hls
> 
> `Referer` header cannot be set in browsers manually as browsers restrict it, so have to find a way to do it from the background process (main process/native http)

Huh?
From what I remember you can set the referer of a request using JS no problem.
And from what I remember a player can play a stream of data, so you can use JS to feed the video to the player.

**Comment by zyrouge** [comment]
> > > what about this player? https://dplayer.js.org/guide.html#hls
> > 
> > 
> > `Referer` header cannot be set in browsers manually as browsers restrict it, so have to find a way to do it from the background process (main process/native http)
> 
> Huh?
> From what I remember you can set the referer of a request using JS no problem.
> And from what I remember a player can play a stream of data, so you can use JS to feed the video to the player.

Refer https://developer.mozilla.org/en-US/docs/Glossary/Forbidden_header_name
